src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenForClientParameterBuilder.cs

```diff
@@ -79,10 +79,13 @@ public AcquireTokenForClientParameterBuilder WithSendX5C(bool withSendX5C)
         /// By default the value for this variable is false.
         /// See https://aka.ms/msal-net-region-discovery for more details.
         /// </summary>
-        /// <param name="autoDetectRegion"><c>true</c> if the token request should be sent to regional ESTS. Otherwise <c>false</c>.
-        /// The default is <c>false</c></param>
+        /// <param name="autoDetectRegion"><c>true</c> if the token request should be sent to regional ESTS. The default is <c>false</c>.
+        /// </param>
+        /// <param name="useRegion"> optional parameter to provide region to MSAL. This parameter will be used along with auto detection of region.
+        /// If the region is auto detected, the provided region will be compared with the detected region and used in telemetry to do analysis on correctness of the region provided.
+        /// If auto region detection fails, the provided region will be used for instance metadata.</param>
         /// <returns>The builder to chain the .With methods</returns>
-        public AcquireTokenForClientParameterBuilder WithAzureRegion(bool autoDetectRegion)
+        public AcquireTokenForClientParameterBuilder WithAzureRegion(bool autoDetectRegion, string useRegion = "")
```

Comment:
Makes sense, regionToUse sounds good. I will update it.

Reply:
Updated

Reply:
i'm a little confused...so sorry for the weird questions...so if I do `.WithAzureRegion()` does that mean I am expecting to use the regional endpoint? If there a reason to even have the boolean? If a customer includes `.WithAzureRegion()` can we assume they want the regional endpoint? so if a customer uses this, it's always going to regional, and then they can specify the actual region they want, if they know it. It just seems counterintuitive to say I want to use the regional endpoint, and then also have to set a boolean to true. interested in discussing offline if you want.

---
In reply to: [549874926](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2315#discussion_r549874926) [](ancestors = 549874926)

Reply:
Initially, there was no plan to ask user to provide the region. But since now adopting regional is a resilience feature, we are adding things like allow the user to pass region as currently we cannot detect region on all platforms. So, the bool `autoDetectRegion` is what currently exists and people are testing it on their test environments. `regionToUse` is an optional field that someone may choose to provide and we will use it in case we are unable to auto detect the region to give more confidence to people adopting it.

---
In reply to: [551514796](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2315#discussion_r551514796) [](ancestors = 551514796,549874926)

Reply:
right, that makes sense. i guess what i'm suggesting is simplifying the API to `public AcquireTokenForClientParameterBuilder WithAzureRegion(string regionToUse = "")` because if they use this builder, they are automatically opting in to regional detection, but if they provide a value, we 1) try that region 2) if that fails try to detect the region 3) use global endpoint

---
In reply to: [551575530](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2315#discussion_r551575530) [](ancestors = 551575530,551514796,549874926)

Reply:
I think you should revert back to what you had before for the reason mentioned above. the developer experience is better that way as it avoids the need for if statements

---
In reply to: [550357996](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2315#discussion_r550357996) [](ancestors = 550357996)